core/src/main/java/org/elasticsearch/index/query/QueryStringQueryBuilder.java

```diff
@@ -326,98 +461,282 @@ public QueryStringQueryBuilder escape(boolean escape) {
+        builder.field("query", this.queryString);
+        if (this.defaultField != null) {
+            builder.field("default_field", this.defaultField);
         }
-        if (fields != null) {
-            builder.startArray("fields");
-            for (String field : fields) {
-                if (fieldsBoosts != null && fieldsBoosts.containsKey(field)) {
-                    field += "^" + fieldsBoosts.get(field);
-                }
-                builder.value(field);
-            }
-            builder.endArray();
+        builder.startArray("fields");
+        for (Map.Entry<String, Float> fieldEntry : this.fieldsAndWeights.entrySet()) {
+            builder.value(fieldEntry.getKey() + "^" + fieldEntry.getValue());
         }
-        if (useDisMax != null) {
-            builder.field("use_dis_max", useDisMax);
+        builder.endArray();
+        builder.field("use_dis_max", this.useDisMax);
+        builder.field("tie_breaker", this.tieBreaker);
+        builder.field("default_operator", this.defaultOperator.name().toLowerCase(Locale.ROOT));
+        if (this.analyzer != null) {
+            builder.field("analyzer", this.analyzer);
         }
-        if (tieBreaker != -1) {
-            builder.field("tie_breaker", tieBreaker);
+        if (this.quoteAnalyzer != null) {
+            builder.field("quote_analyzer", this.quoteAnalyzer);
         }
-        if (defaultOperator != null) {
-            builder.field("default_operator", defaultOperator.name().toLowerCase(Locale.ROOT));
+        builder.field("auto_generate_phrase_queries", this.autoGeneratePhraseQueries);
+        builder.field("max_determinized_states", this.maxDeterminizedStates);
+        if (this.allowLeadingWildcard != null) {
+            builder.field("allow_leading_wildcard", this.allowLeadingWildcard);
         }
-        if (analyzer != null) {
-            builder.field("analyzer", analyzer);
+        builder.field("lowercase_expanded_terms", this.lowercaseExpandedTerms);
+        builder.field("enable_position_increments", this.enablePositionIncrements);
+        this.fuzziness.toXContent(builder, params);
+        builder.field("fuzzy_prefix_length", this.fuzzyPrefixLength);
+        builder.field("fuzzy_max_expansions", this.fuzzyMaxExpansions);
+        if (this.fuzzyRewrite != null) {
+            builder.field("fuzzy_rewrite", this.fuzzyRewrite);
         }
-        if (quoteAnalyzer != null) {
-            builder.field("quote_analyzer", quoteAnalyzer);
-        }
-        if (autoGeneratePhraseQueries != null) {
-            builder.field("auto_generate_phrase_queries", autoGeneratePhraseQueries);
+        builder.field("phrase_slop", this.phraseSlop);
+        if (this.analyzeWildcard != null) {
+            builder.field("analyze_wildcard", this.analyzeWildcard);
         }
-        if (maxDeterminizedStates != null) {
-            builder.field("max_determinized_states", maxDeterminizedStates);
+        if (this.rewrite != null) {
```

Comment:
only few stayed Boolean, I moved everything I could to primitive type. Exceptions are `analyzeWildcard` and `allowLeadingWildcard`. `null` there has a meaning (see toQuery method), as the value needs to be taken from the context, which we have only in toQuery, so we can't use it as a default value instead...